components/Search.tsx

```diff
@@ -7,10 +7,17 @@ import FormationCard from "@/components/FormationCard";
 import { ScrollArea } from "@/components/ui/scroll-area";
 import { Separator } from "@/components/ui/separator";
 import { Label } from "@/components/ui/label";
+import { FormationData } from "@/lib/formations";
 
-export default function Search({ cmsData }: { cmsData: any }) {
+export default function Search({
+  cmsData,
+  prePopulated,
+}: {
+  cmsData: any;
+  prePopulated: FormationData[];
+}) {
   const [search, setSearch] = useState("");
-  const [results, setResults] = useState([]);
+  const [results, setResults] = useState(prePopulated);
   const [loading, setLoading] = useState(false);
 
   const handleSearch = async (e: FormEvent) => {
```

Comment:
Again not something changed with this Pr, but i wonder if we should try utilise react query for our data fetching
https://tanstack.com/query/latest/docs/framework/react/overview

Out the box retries, reattempts and other cool functions

Reply:
Using react query is something to consider, I’m personally not totally sold on it for nextjs projects.

it’s mostly for client components, and has a totally different caching pattern than next’s fetch api.

im not opposed to it per say but I just haven’t found a great usecase for it myself